templates/data/_form-data-spark.html

```diff
@@ -0,0 +1,111 @@
+                    <input required id="firstName" name="firstName" maxlength="255" type="text" required="required" />
+                  </li>
+                  <li class="p-list__item">
+                    <label for="lastName">Last name:</label>
+                    <input required id="lastName" name="lastName" maxlength="255" type="text" required="required" />
+                  </li>
+                  <li class="p-list__item">
+                    <label for="company">Company name:</label>
+                    <input required id="company" name="company" maxlength="255" type="text">
+                  </li>
+                  <li class="p-list__item">
+                    <label for="title">Job title:</label>
+                    <input required id="title" name="title" maxlength="255" type="text">
+                  </li>
+                  <li class="p-list__item">
+                    <label for="email">Work email:</label>
+                    <input required id="email" name="email" maxlength="255" type="email" pattern="^[^ ]+@[^ ]+\.[a-z]{2,26}$" required="required" />
+                  </li>
+                  <li class="p-list__item">
+                    <label for="phone">Mobile/cell phone number:</label>
+                    <input required id="phone" name="phone" maxlength="255" type="tel" required="required" />
+                  </li>
+                  <li class="p-list__item">
+                    <label class="p-checkbox">
+                      <input class="p-checkbox__input" value="yes" aria-labelledby="canonicalUpdatesOptIn" name="canonicalUpdatesOptIn" type="checkbox">
+                      <span class="p-checkbox__label" id="canonicalUpdatesOptIn">I agree to receive information about Canonical's products and services.</span>
+                    </label>
+                  </li>
+                  <li class="p-list__item">In submitting this form, I confirm that I have read and agree to Canonical's <a href="/legal/terms-and-policies/privacy-policy">Privacy Policy</a>.</li>
+                  {# These are honey pot fields to catch bots #}
+                  <li class="u-off-screen">
+                    <label class="website" for="website">Website:</label>
+                    <input name="website" type="text" class="website" autocomplete="off" value="" id="website" tabindex="-1" />
+                  </li>
+                  <li class="u-off-screen">
+                    <label class="name" for="name">Name:</label>
+                    <input name="name" type="text" class="name" autocomplete="off" value="" id="name" tabindex="-1" />
+                  </li>
+                  {# End of honey pots #}
+                </ul>
+    
+                <div class="u-hide">
+                  <input type="hidden" aria-hidden="true" aria-label="hidden field" name="returnURL" value="{{ returnURL }}" />
+                  <input type="hidden" aria-hidden="true" aria-label="hidden field" name="formid" value="5505" />
```

Comment:
You are passing a form_id and hardcoding it here. I think the form id should be `4908`. Found on the ubuntu.com/data/spark page.

Reply:
As mentioned by Juan it's 5505. They have changed the marketo ID for c.com, 4908 is the old one.

Yes, we shouldn't pass form_id, because form fields and some text change from page to page. I'll remove it.